test/testdata/inferencepool-e2e.yaml

```diff
@@ -0,0 +1,126 @@
+apiVersion: inference.networking.x-k8s.io/v1alpha2
```

Comment:
This looks like a duplication of the existing inferencepool-resources.yaml, excluding the namespace field.
I would prefer if we can remove the namespace field from all resources in the `inferencepool-resources.yaml` file and specify the namespace in the e2e test (as env var as specified in other comment).

there is a big value in testing e2e our "Getting Started" files, so we are self testing the public documentation. 
I would avoid having duplications as much as possible.

Reply:
I suppose I could use a similar approach to the one you described above. However, that might complicate the getting started guide? How would the user of the guide make sure the env var was parsed in the file?

Reply:
in the e2e a similar approach would work. in getting started guide I’m not sure we have an agreement of the other maintainers about being able to choose the namespace (I’m in favor of it).
if there is an agreement, you can use envsubst command to replace env var with it’s value. 

for example - 
envsubst < myfile.yaml | kubectl apply -f - 

this discussion is relevant only for the subject serviceaccount in the ClusterRoleBinding, as the namespace in the Service and Deployment resources can be juat removed and then you can decide the ns during e2e test.

Reply:
Should we tackle this in a follow-up then? If we change the getting started file in this PR its going to expand the scope considerably as we now need to update the guide with the envvar substitution command.

cc @danehans

Reply:
>If we change the getting started file in this PR

+1 on not changing the getting started guide in this PR.

Reply:
> there is a big value in testing e2e our "Getting Started" files, so we are self testing the public documentation.

I agree with this point. We may need to consider 2 different e2e tests, 1 for testing the quickstart and another that is more programmatic to support various inference use cases. Maybe we track this in a separate issue?

cc: @ahg-g @kfswain

Reply:
I prefer to move away from yaml files and just create those objects in code,,, the envoy one is complex, so I don't mind it, but other than that I think it is more manageable to have the test create those objects in code.

Reply:
oh, well, creating the pool requires also creating all the epp and the rbac resources, so it is also complex...